frontend/packages/core/src/Input/select.tsx

```diff
@@ -241,8 +249,18 @@ export const Select = ({
     if (!value) {
       return;
     }
-    setSelectedIdx(flatOptions.findIndex(opt => opt?.value === value || opt?.label === value));
-    onChange && onChange(value);
+
+    if (multiple) {
+      const newSelectedIdxs = value.map(val =>
+        flatOptions.findIndex(opt => opt.value === val || opt.label === val)
+      );
+      setSelectedIdxs(newSelectedIdxs);
+    } else {
+      const index = flatOptions.findIndex(opt => opt.value === value || opt.label === value);
+      setSelectedIdxs([index]);
+    }
```

Comment:
We can probably simplify this a bit:

```suggestion
const findIndex = (val) => flatOptions.findIndex(opt => opt.value === val || opt.label === val);
setSelectedIdxs(multiple ? value.map(val => findIndex(val)) : [findIndex(value)]);
```